### PR TITLE
Doc: fix Webots version

### LINKED ISSUE #611: Online documentation still referring to R2019a
Not sure if this is a temporary issue until R2019b is released, but in the R2019b documentation page there are references to R2019a

https://www.cyberbotics.com/doc/reference/index?version=R2019b
Screenshot from 2019-06-19 11-59-27

### PR DESCRIPTION
**Pull request by fabienrohrer**
Fix #611

### CODE REVIEW

**Comment by omichel** [comment]
Normally, this [script](https://github.com/omichel/webots/blob/develop/src/bin/new_version.sh) should have updated the version in this `showdown-extensions.js` properly. Do you know why it didn't work?

**Comment by fabienrohrer** [comment]
The script is buggy, I'm just trying to fix it.

**Comment by fabienrohrer** [comment]
I tested `new_version.sh` a while on Ubuntu 18, and it is working like a charm. I can upgrade files when upgrading the Webots version from major to minor and the contrary.
`showdow-extensions.js` is successfully modified.

Regarding to the [history of this file](https://github.com/omichel/webots/commits/develop/docs/js/showdown-extensions.js), I can only conclude on a mistake during this commit for now. We should pay attention at this when doing the next update.

I improved the script to fail in case it is used on macOS or Windows where it's far to work.

**Comment by fabienrohrer** [comment]
👍

**Comment by omichel** [comment]
I believe this script works on Windows. It seems that it failed because in `showdown-extensions.js` the `full` string was wrong, it was `R2019a` whereas the script expected `R2019a revision 1`. Therefore nothing was replaced. I believe this script is wrong, it should inform the user if it failed at modifying a file or it should be more robust, e.g., replace `full: '*'` with `full: '$new_version'`, instead of searching for a previous version (which was missing in this case).

**Comment by fabienrohrer** [comment]
I clearly missed something, let me check again..

File: src/bin/new_version_file.sh [hunk 1]
```diff
@@ -6,6 +6,10 @@ replace_string="${escaped_string//\\s/ }"
 echo "s/"$1"/"$replace_string"/g" > sed.cmd
 sed -e 's/@/ /g' sed.cmd > sed2.cmd
 sed -f sed2.cmd  $3 > tmp.bak
+if ! cmp tmp.bak $3 >/dev/null 2>&1
+then
+  >&2 echo "Replacement failed in '$3'"
+fi
```
Comment (omichel) [inline src/bin/new_version_file.sh]: Very good!

**Comment by fabienrohrer** [comment]
👍

**Comment by omichel** [comment]
:eyes:

**Comment by omichel** [comment]
It doesn't seems to work:
```
$ ./new_version.sh R2019b 0 R2019b 1
Update application and documentation version..
Replacement failed in '../webots/core/WbApplicationInfo.cpp'
Replacement failed in '../../resources/version.txt'
Replacement failed in '../packaging/webots_version.txt'
Replacement failed in '../../Contents/Info.plist'
Replacement failed in '../../docs/js/showdown-extensions.js'
Replacement failed in '../../docs/js/showdown-extensions.js'
$ git diff ../webots/core/WbApplicationInfo.cpp
diff --git a/src/webots/core/WbApplicationInfo.cpp b/src/webots/core/WbApplicationInfo.cpp
index 475ef267..13947f79 100644
--- a/src/webots/core/WbApplicationInfo.cpp
+++ b/src/webots/core/WbApplicationInfo.cpp
@@ -24,7 +24,7 @@ const WbVersion &WbApplicationInfo::version() {
   static bool firstCall = true;

   if (firstCall) {
-    static QString webotsVersionString = "R2019b";  // updated by script
+    static QString webotsVersionString = "R2019b revision 1";  // updated by script
     bool success = webotsVersion.fromString(webotsVersionString);
     if (!success)
       WbLog::fatal(QObject::tr("Internal error: the Webots version is not computable."));
```

**Comment by fabienrohrer** [comment]
I'm checking on Windows.

**Comment by omichel** [comment]
I committed a partial fix, but I still get this:
```
$ ./new_version.sh R2019b 0 R2019b 1
Update application and documentation version..
Replacement failed in '../../Contents/Info.plist'
Replacement failed in '../../docs/js/showdown-extensions.js'
Replacement failed in '../../docs/js/showdown-extensions.js'
```
Which is normal because in these case, we do not necessarily modify the file (e.g., the copyright date may not need to be changed).

**Comment by fabienrohrer** [comment]
Ok, thank you.

Not sure what to do now.

Maybe that rather displaying an error, we could simply display a message if the file has changed or not. It would be on the human who called the script to check if the modification is pertinent or not..

**Comment by omichel** [comment]
I have an idea, let me commit something...

**Comment by fabienrohrer** [comment]
Ok great.

**Comment by omichel** [comment]
👍

**Comment by DavidMansolino** [comment]
:+1:

**Comment by DavidMansolino** [comment]
:eyes:

**Comment by fabienrohrer** [comment]
Thank you to have completed this \^^./

**Pull request merged**